Content/NPCs/Corruption/SoulNibbler.cs

```diff
@@ -0,0 +1,155 @@
+using Terraria.ID;
+
+namespace StarlightRiver.Content.NPCs.Corruption
+{
+	internal class SoulNibbler : Swarmer
+	{
+		public ref float Timer => ref NPC.ai[0];
+		public ref float State => ref NPC.ai[1];
+
+		public override string Texture => "StarlightRiver/Assets/NPCs/Corruption/" + Name;
+
+		public override float IdealDistance => 24;
+
+		public override float PushPower => 0.7f;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Soul Nibbler");
+		}
+
+		public override void SetDefaults()
+		{
+			NPC.width = 32;
+			NPC.height = 34;
+			NPC.damage = 8;
+			NPC.lifeMax = 34;
+			NPC.knockBackResist = 1.1f;
+			NPC.noGravity = true;
+			NPC.aiStyle = -1;
+
+			NPC.HitSound = SoundID.NPCHit1;
+			NPC.DeathSound = SoundID.NPCDeath1;
+		}
+
+		public override void SetBestiary(BestiaryDatabase database, BestiaryEntry bestiaryEntry)
+		{
+			bestiaryEntry.Info.AddRange(new IBestiaryInfoElement[]
+			{
+				BestiaryDatabaseNPCsPopulator.CommonTags.SpawnConditions.Biomes.TheCorruption,
+				BestiaryDatabaseNPCsPopulator.CommonTags.SpawnConditions.Times.NightTime,
+				new FlavorTextBestiaryInfoElement("These tiny pests take after their larger bretheren, floating eerily and swarming any foolish enough to come near their tiny, but awfully sharp, fangs.")
+			});
+		}
+
+		public override void AI()
+		{
+			Timer++;
+
+			if (Main.dayTime)
+			{
+				NPC.noTileCollide = true;
+				NPC.velocity.Y += 0.25f;
+
+				return;
+			}
+
+			if (NPC.target >= 0)
+			{
+				Player target = Main.player[NPC.target];
+				NPC.rotation = NPC.DirectionTo(target.Center).ToRotation() - 1.57f;
+			}
+
+			if (State == 0)
+			{
+				ShoveSwarm();
+				NPC.TargetClosest();
+
+				if (NPC.target >= 0)
+				{
+					Player target = Main.player[NPC.target];
+
+					NPC.velocity += NPC.DirectionTo(target.Center) * 0.05f;
+
+					if (NPC.velocity.LengthSquared() > 36)
+						NPC.velocity = Vector2.Normalize(NPC.velocity) * 5.99f;
+
+					if (Timer > 90)
+						Lighting.AddLight(NPC.Center, new Vector3(0.3f, 0.4f, 0f) * (Timer - 90) / 30f);
+
+					if (Timer > 120)
+					{
+						State = 1;
+						Timer = 0;
+					}
+				}
+			}
+
+			if (State == 1)
+			{
+				if (Timer == 10 && NPC.target >= 0)
+				{
+					Player target = Main.player[NPC.target];
+					NPC.velocity += NPC.DirectionTo(target.Center) * 8f;
+
+					for (int k = 0; k < 10; k++)
+						Dust.NewDustPerfect(NPC.Center, DustID.GreenBlood, Main.rand.NextVector2Circular(2, 2));
+				}
+
+				Dust.NewDustPerfect(NPC.Center, DustID.GreenBlood, Main.rand.NextVector2Circular(1, 1), 0, default, 0.25f);
+				Lighting.AddLight(NPC.Center, new Vector3(0.3f, 0.4f, 0f) * (1f - Timer / 60f));
+
+				if (Timer >= 60)
+				{
+					State = 0;
+					Timer = 0;
+				}
+			}
+		}
+
+		public override bool CanHitPlayer(Player target, ref int cooldownSlot)
+		{
+			return State == 1 && Timer > 10;
+		}
+
+		public override void OnHitPlayer(Player target, Player.HurtInfo hurtInfo)
+		{
+			NPC.velocity *= -1;
```

Comment:
may be outside the scope of this PR but OnHitPlayer only runs for the client being hit

Reply:
just toss a //sync TODO: or something for now